pandas/plotting/_core.py

```diff
@@ -1706,11 +1706,15 @@ def _plot(data, x=None, y=None, subplots=False,
             if x is not None:
                 if is_integer(x) and not data.columns.holds_integer():
                     x = data.columns[x]
+                elif not isinstance(data[x], Series):
```

Comment:
```
from pandas.core.dtypes.generic import ABCSeries, ABCDataFrame
```

``not isinstance(data[x], ABCSeries)``

and change the ``DataFrame`` tests to use ``ABCDataFrame`` (and remove the inline import)

Reply:
> and change the DataFrame tests to use ABCDataFrame (and remove the inline import)

Not sure what you mean here? I changed the test to use `ABCDataFrame` but that doesn't seem right?